### PR TITLE
Migrate to Firebase common executors

### PR DESCRIPTION
**Pull request by mrober**
Migrate to Firebase common executors. Use the new `CrashlyticsWorker` for common and disk write workers.

### CODE REVIEW

File: firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/CrashlyticsTestCase.java [hunk 1]
```diff
@@ -21,6 +21,8 @@
 
 /** Base class for all our TestCases. */
 public class CrashlyticsTestCase extends TestCase {
+  // TODO(mrober): Setup the workers with test executors in here?
```
Comment (themiswang) [inline firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/CrashlyticsTestCase.java:24]: This is something we want to do next?

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsCore.java [hunk 2]
```diff
@@ -108,21 +109,22 @@ public CrashlyticsCore(
       BreadcrumbSource breadcrumbSource,
       AnalyticsEventLogger analyticsEventLogger,
       FileStore fileStore,
-      ExecutorService crashHandlerExecutor,
       CrashlyticsAppQualitySessionsSubscriber sessionsSubscriber,
-      RemoteConfigDeferredProxy remoteConfigDeferredProxy) {
+      RemoteConfigDeferredProxy remoteConfigDeferredProxy,
+      CrashlyticsWorker commonWorker,
```
Comment (themiswang) [inline firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsCore.java:114]: I was thinking about pass firebase common background here then init crashlytics work within the core initialization.

File: firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/metadata/MetaDataStoreTest.java [hunk 3]
```diff
@@ -104,9 +104,10 @@ public void testWriteUserData_null() {
     assertNull(userData.getUserId());
   }
 
-  public void testWriteUserData_emptyString() {
+  public void testWriteUserData_emptyString() throws Exception {
```
Comment (themiswang) [inline firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/metadata/MetaDataStoreTest.java:107]: Oh test I have fixed in my pr, I think you can ignore the failed unit tests in metatDataStoreTest.

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsCore.java (hunk 2 above)
Comment (mrober) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsCore.java:114]: We use common worker for settings in this class.

File: firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/CrashlyticsTestCase.java (hunk 1 above)
Comment (mrober) [reply firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/CrashlyticsTestCase.java:24]: Discussed offline. We want to have more flexibility in individual test cases, then some general test structure that does things for you behind the scene. So we won't do this.

File: firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/metadata/MetaDataStoreTest.java (hunk 3 above)
Comment (mrober) [reply firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/metadata/MetaDataStoreTest.java:107]: Consider it ignored.

**Review by themiswang — APPROVED** [review]

**Comment by mrober** [comment]
#4345

**Pull request merged**